src/spikeinterface/core/baserecording.py

```diff
@@ -331,6 +331,8 @@ def get_traces(
         segment_index = self._check_segment_index(segment_index)
         channel_indices = self.ids_to_indices(channel_ids, prefer_slice=True)
         rs = self._recording_segments[segment_index]
+        start_frame = int(max(0, start_frame)) if start_frame is not None else 0
```

Comment:
I think you guys wanted to raise an error if start_frame was lower than 0? 

https://github.com/SpikeInterface/spikeinterface/issues/1989

Reply:
done. Also a check that start frame cannot be greater than end_frame

Reply:
What about the end_frame > num_samples?

Reply:
Yep, I reread the thread and I'll add that on Monday! Might need to tweak some failing tests then

Reply:
Maybe it would be beter to leave this like this and then add that condition and fix the tests on another PR? 

I also think that you will get failures that will bow you down and this is and improvement already as it is.

Reply:
Great idea!